glean-core/src/upload/request.rs

```diff
@@ -155,19 +158,25 @@ impl Builder {
     ///
     /// This method will panic if any of the required fields are missing:
     /// `document_id`, `path` and `body`.
-    pub fn build(self) -> PingRequest {
-        PingRequest {
+    pub fn build(self) -> Result<PingRequest> {
+        let body = self
+            .body
+            .expect("body must be set before attempting to build PingRequest");
+
+        if body.len() > self.body_max_size {
+            return Err(ErrorKind::PingBodyOverflow(body.len() as u64).into());
```

Comment:
What's the reason we don't keep this as a `usize`?

Reply:
Hm, no reason. I'll change it.